src/librustc_middle/ty/layout.rs

```diff
@@ -2368,11 +2369,55 @@ where
         sig: ty::PolyFnSig<'tcx>,
         extra_args: &[Ty<'tcx>],
         caller_location: Option<Ty<'tcx>>,
+        codegen_fn_attr_flags: CodegenFnAttrFlags,
         mk_arg_type: impl Fn(Ty<'tcx>, Option<usize>) -> ArgAbi<'tcx, Ty<'tcx>>,
     ) -> Self;
     fn adjust_for_abi(&mut self, cx: &C, abi: SpecAbi);
 }
 
+fn can_unwind(
+    panic_strategy: PanicStrategy,
+    codegen_fn_attr_flags: CodegenFnAttrFlags,
+    abi_conv: Conv,
+) -> bool {
```

Comment:
I still think this should probably be nested in `FnAbi::new_internal` (or otherwise have a longer name, but that's a hack around the lack of organization).

Reply:
I'd prefer to have this pulled out as `FnAbit::new_internal` is already quite large and this is very self-contained. How about naming it something like `function_abi_can_unwind`?

Reply:
`fn_can_unwind` should fine, *I guess*.

Reply:
If you feel strongly, I don't mind inlining it that much.

Reply:
It's more that the `TyLayout` stuff and the `FnAbi` stuff really shouldn't be in the same module, it's just that we haven't reorganized since `rustc_target::abi` was created.
Either starting with `fn_` or inlining should be fine.